crates/fuel-core/src/lib.rs

```diff
@@ -63,27 +76,44 @@ pub struct ShutdownListener {
     pub token: CancellationToken,
 }
 
+fn create_signal_futures(
+    signal_variants: Vec<tokio::signal::unix::SignalKind>,
+) -> tokio::io::Result<FuturesUnordered<Pin<Box<dyn Future<Output = Option<()>> + Send>>>>
+{
+    let mut signals = Vec::with_capacity(signal_variants.len());
+    for signal_kind in signal_variants {
+        signals.push(tokio::signal::unix::signal(signal_kind)?);
+    }
+
+    let signal_futs = FuturesUnordered::new();
+
+    for mut signal in signals {
+        signal_futs.push(Box::pin(async move { signal.recv().await })
+            as Pin<Box<dyn Future<Output = Option<()>> + Send>>);
+    }
+
+    Ok(signal_futs)
+}
+
 impl ShutdownListener {
     pub fn spawn() -> Self {
         let token = CancellationToken::new();
         {
             let token = token.clone();
             tokio::spawn(async move {
-                let mut sigterm = tokio::signal::unix::signal(
-                    tokio::signal::unix::SignalKind::terminate(),
-                )?;
-
-                let mut sigint = tokio::signal::unix::signal(
-                    tokio::signal::unix::SignalKind::interrupt(),
-                )?;
                 #[cfg(unix)]
-                tokio::select! {
-                    _ = sigterm.recv() => {
-                        tracing::info!("Received SIGTERM");
-                    }
-                    _ = sigint.recv() => {
-                        tracing::info!("Received SIGINT");
-                    }
+                {
+                    let mut signal_futs = create_signal_futures(vec![
+                        tokio::signal::unix::SignalKind::terminate(),
+                        tokio::signal::unix::SignalKind::interrupt(),
+                        tokio::signal::unix::SignalKind::from_raw(libc::SIGSEGV),
+                        tokio::signal::unix::SignalKind::from_raw(libc::SIGKILL),
+                        tokio::signal::unix::SignalKind::from_raw(libc::SIGABRT),
+                        tokio::signal::unix::SignalKind::hangup(),
+                    ])?;
+
+                    signal_futs.next().await;
+                    tracing::error!("Received shutdown signal");
```

Comment:
```suggestion
                {
                    let signals_to_catch = vec![
                        tokio::signal::unix::SignalKind::terminate(),
                        tokio::signal::unix::SignalKind::interrupt(),
                        tokio::signal::unix::SignalKind::from_raw(libc::SIGSEGV),
                        tokio::signal::unix::SignalKind::from_raw(libc::SIGKILL),
                        tokio::signal::unix::SignalKind::from_raw(libc::SIGABRT),
                        tokio::signal::unix::SignalKind::hangup(),
                    ];
                    let mut signal_futs = create_signal_futures()?;

                    let signal_kind = signal_futs.next().await;
                    tracing::error!("Received shutdown signal: {signal_kind:?}");
```

nit: Let's break the kinds out for some self-documentation.

non-nit: Let's return the kind type. It implements Debug, so we should be able to give more info through the `error!` trace.